system/lib/libc/sbrk.c

```diff
@@ -79,7 +71,7 @@ void *sbrk(intptr_t increment_) {
     // increase the WebAssembly Memory size, and abort if that fails.
     if ((increment > 0 && new_brk <= old_brk)
      || (new_brk > emscripten_get_heap_size() && !emscripten_resize_heap(new_brk))) {
-      SET_ERRNO();
+      errno = ENOMEM;
```

Comment:
I think it would be good to know why this does not affect code size. You mention the two possibilities in the description, I think we can find out which it is by building and inspecting the output? Building a program that does use or not use errno in user code, and comparing those, could also help.

Reply:
I'm not sure this is worth the effort to investigate this much since we don't make any effort to avoid using errno anywhere else in libc.  I will do some measurements though.. since you have peeked my interest.

Reply:
I ran a simple test of a program that just calls malloc with the current `-sSUPPORT_ERRNO` setting:

```
$ cat test.c 
#include <malloc.h>

int main() {
  return (int)malloc(10);
}
$ ./emcc -O2 test.c -o no_errno.js -sSUPPORT_ERRNO=0 -Wl,--trace-symbol=errno
$ ./emcc -O2 test.c
$ du -b a.out.wasm no_errno.wasm 
4248	a.out.wasm
4235	no_errno.wasm
```

It looks like the cost saving of not using errno in malloc is about 13 bytes or 0.3%, but that doesn't show up in any of our `test_minimal_runtime_code_size` because they all use emmalloc which already doesn't set errno AFIACT.

I don't think those 13 byte are actually going to be saved in any real program since `errno` will almost always be included natively unless we do a bunch of work on libc to remove it.